lib/instrumentation/modules/aws-sdk/sqs.js

```diff
@@ -130,7 +130,7 @@ function shouldIgnoreRequest (request, agent) {
     }
   }
 
-  if (!agent.currentTransaction) {
+  if (!agent.currentTransaction || agent.currentTransaction.ended) {
```

Comment:
@astorm What about something like this instead:

```
--- a/lib/instrumentation/modules/aws-sdk/sqs.js
+++ b/lib/instrumentation/modules/aws-sdk/sqs.js
@@ -152,6 +152,10 @@ function instrumentationSqs (orig, origArguments, request, AWS, agent, { version
   const action = getActionFromRequest(request)
   const name = getSpanNameFromRequest(request)
   const span = agent.startSpan(name, type, subtype, action)
+  if (!span) {
+    return orig.apply(request, origArguments)
+  }
+
   span.setDestinationContext(getMessageDestinationContextFromRequest(request))
   span.setMessageContext(getMessageContextFromRequest(request))
```

This is the typical technique I see in the other instrumentations, e.g.:

```
tedious.js
72:        const span = agent.startSpan(null, 'db', 'mssql', 'query')
73-        if (!span) {
74-          return super.makeRequest(...arguments)
75-        }

cassandra-driver.js
45:      const span = agent.startSpan('Cassandra: Connect', 'db', 'cassandra', 'connect')
46-      if (!span) {
47-        return original.apply(this, arguments)
48-      }

mongodb.js
44:    const span = agent.startSpan(name, 'db', 'mongodb', 'query')
45-    if (span) {
46-      activeSpans.set(event.requestId, span)
```

That allows keeping the logic for whether a span could be created inside `agent.startSpan()` instead of duplicating that logic.

Reply:
_nod_ -- change made.  We're now checking to see if a span is successfully created after calling startSpan, and if not bailing out by calling the original method.  also, dings the "something to consider for a custom linting rule at a future date" bell 🛎️ .

Reply:
>  also, dings the "something to consider for a custom linting rule at a future date" bell 🛎️ .

We could re-implement the agent in Rust which would require us to handle the `Option<Span>`. ;)